lib/matplotlib/collections.py

```diff
@@ -833,6 +841,35 @@ def set_edgecolor(self, c):
         self._original_edgecolor = c
         self._set_edgecolor(c)
 
+    def _set_hatchcolor(self, c):
+        c = mpl._val_or_rc(c, "hatch.color")
+        if c == "inherit":
+            if self._original_edgecolor is not None:
+                c = self._original_edgecolor
+            else:
+                c = mpl.rcParams["patch.edgecolor"]
+
+        if isinstance(c, str) and c in ('none', 'face'):
+            c = 'black'
```

Comment:
For now, It is intended that "inherit" is immediately overwritten with the expected color. As for defaulting to the rcparam, it is possible to use `patch.edgecolor` as a default fallback value here as well.

```diff
--- a/lib/matplotlib/collections.py
+++ b/lib/matplotlib/collections.py
@@ -846,12 +846,11 @@ class Collection(artist.Artist, cm.ScalarMappable):
         if c == "inherit":
             if self._original_edgecolor is not None:
                 c = self._original_edgecolor
+                if isinstance(c, str) and c in ('none', 'face'):
+                    c = mpl.rcParams["patch.edgecolor"]
             else:
                 c = mpl.rcParams["patch.edgecolor"]
 
-        if isinstance(c, str) and c in ('none', 'face'):
-            c = 'black'
-
         self._hatchcolors = mcolors.to_rgba_array(c, self._alpha)
         self.stale = True
 
```

Also, I think it would be better to move the if condition up, as that is the only scope where `c` could get a value of 'none' or 'face'.